improver_tests/calibration/ensemble_calibration/test_ApplyEMOS.py

```diff
@@ -265,6 +267,69 @@ def test_land_sea_mask(self):
+        distribution) for the truncated normal distribution."""
+        coefficients = iris.cube.CubeList([])
+        for cube in self.coefficients:
+            cube.attributes["distribution"] = "truncnorm"
+            cube.attributes["shape_parameters"] = np.array([0, np.inf], np.float32)
+            coefficients.append(cube)
+
+        expected_mean = np.mean(self.percentiles.data)
+        expected_data = np.array(
+            [
+                np.full((3, 3), 10.265101),
+                np.full((3, 3), 10.4),
+                np.full((3, 3), 10.534898),
+            ]
+        )
+        result = ApplyEMOS()(self.percentiles, coefficients, realizations_count=3)
+        self.assertIn("percentile", get_dim_coord_names(result))
+        self.assertArrayAlmostEqual(result.data, expected_data)
+        self.assertAlmostEqual(np.mean(result.data), expected_mean)
+
+    def test_null_percentiles_truncnorm_alternative_shape_parameters(self):
+        """Test effect of "neutral" emos coefficients in percentile space
+        (this is small but non-zero due to limited sampling of the
+        distribution) for the truncated normal distribution with alternative
+        shape parameters to show the truncnorm distribution having an effect."""
+        coefficients = iris.cube.CubeList([])
+        for cube in self.coefficients:
+            cube.attributes["distribution"] = "truncnorm"
+            cube.attributes["shape_parameters"] = np.array([10, np.inf], np.float32)
+            coefficients.append(cube)
+
+        expected_mean = np.mean(self.percentiles.data)
+        expected_data = np.array(
+            [
+                np.full((3, 3), 10.275656),
+                np.full((3, 3), 10.405704),
+                np.full((3, 3), 10.5385),
+            ]
+        )
+        result = ApplyEMOS()(self.percentiles, coefficients, realizations_count=3)
+        self.assertIn("percentile", get_dim_coord_names(result))
+        self.assertArrayAlmostEqual(result.data, expected_data)
+        self.assertNotAlmostEqual(np.mean(result.data), expected_mean)
+
+    def test_invalid_attribute(self):
+        """Test that an exception is raised if multiple different distribution
+        attributes are provided within the coefficients cubelist."""
+        self.coefficients[0].attributes["distribution"] = "truncnorm"
+        msg = "Coefficients must share the same"
+        with self.assertRaisesRegex(ValueError, msg):
+            ApplyEMOS()(self.percentiles, self.coefficients, realizations_count=3)
+
+    def test_missing_attribute(self):
+        """Test that an exception is raised if the expected distribution
+        attribute is missing from within the coefficients cubelist."""
+        self.coefficients[0].attributes.pop("distribution")
+        msg = "Coefficients must share the same"
+        with self.assertRaisesRegex(ValueError, msg):
+            ApplyEMOS()(self.percentiles, self.coefficients, realizations_count=3)
```

Comment:
As I commented above, there is no handling here of what happens if the attribute is just outright missing on all input cubes. It only checks for consistency at the moment.

Reply:
I've added a test to cover the complete lack of an expected attribute on the coefficients cubes.